### PR TITLE
Rejectfeature

### PR DESCRIPTION
**Pull request by jsidd**
Pull Request for new reject feature
Connected to #238

### CODE REVIEW

File: app/mailers/notifications.rb [hunk 1]
```diff
@@ -9,11 +9,27 @@ def import_approved(import)
       :subject => "Your Safecast import has been approved")
   end
 
+  def import_rejected(import)
+    @import = import
+    mail(
```
Comment (matschaffer) [inline app/mailers/notifications.rb]: Might be good to reformat these at some point. The indentation & parens isn't very idomatic. Nbd for this PR though.

File: db/structure.sql [hunk 2]
```diff
@@ -415,7 +419,9 @@ CREATE TABLE measurement_imports (
     orientation character varying(255),
     cities text,
     comment character varying(255),
-    subtype measurement_imports_subtype DEFAULT 'None'::measurement_imports_subtype
+    subtype measurement_imports_subtype DEFAULT 'None'::measurement_imports_subtype,
+    rejected boolean DEFAULT false,
+    rejected_by character varying(255)
```
Comment (matschaffer) [inline db/structure.sql]: This should be a user id rather than a name. Will require some re-work.

File: app/models/bgeigie_import.rb [hunk 3]
```diff
@@ -81,6 +85,21 @@ def approve!
     Notifications.import_approved(self).deliver
   end
 
+  def reject!
+    self.update_column(:rejected, true)
+    self.update_column(:rejected_by, user.email)
```
Comment (matschaffer) [inline app/models/bgeigie_import.rb]: Should make this a user id as well.

File: app/views/bgeigie_imports/_status.html.erb [hunk 4]
```diff
@@ -10,4 +10,9 @@
   <span class="label label-info">
     <%= @bgeigie_import.status.humanize %>
   </span>
-<%- end -%>
\ No newline at end of file
+<%- end -%>
+<%- if @bgeigie_import.rejected == true -%>
+  <span class="label label-warning">
+    Rejected by: <%= @bgeigie_import.rejected_by %>
```
Comment (matschaffer) [inline app/views/bgeigie_imports/_status.html.erb]: Once this is a user reference, we can make this `rejected_by.email`

File: app/views/bgeigie_imports/index.html.erb [hunk 5]
```diff
@@ -58,6 +66,7 @@
       <%= table_sort_header(:bgeigie_import, :source) %>
       <%= table_sort_header(:bgeigie_import, :measurements_count) %>
       <%= table_sort_header(:bgeigie_import, :status) %>
+      <%= table_sort_header(:bgeigie_import, :rejected_by) %>
```
Comment (matschaffer) [inline app/views/bgeigie_imports/index.html.erb]: Not sure this really needs to be in the index. Was there an ask for it?

**Comment by matschaffer** [comment]
@jsidd what's the latest on this? Does it just need a rebase or are there still some changes you'd like to make?

**Comment by matschaffer** [comment]
Merging now since this has already been applied to production

**Comment by eitoball** [comment]
> Merging now since this has already been applied to production

What do you mean? I am curious.

**Comment by matschaffer** [comment]
We had a deployment miscommunication where this went to prod (including
migrations) instead of staging.

Rather than have to deal with potential migration changes I opted to merge
and we can address issues in additional PRs.

On Wednesday, 15 June 2016, Eito Katagiri <notifications@github.com> wrote:

> Merging now since this has already been applied to production
>
> What do you mean? I am curious.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/Safecast/safecastapi/pull/260#issuecomment-226151780>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAACskglVtShiex0DvelAQdKmrfx5IaCks5qL9ctgaJpZM4IsO1L>
> .
>


-- 

-Mat

matschaffer.com

**Pull request merged**